content/src/view.js

```diff
@@ -2208,44 +2208,15 @@ function changeEditorText(paneState, text) {
   if (text.length && text.charAt(text.length - 1) != '\n') {
     text += '\n';
   }
-  paneState.changeHandler.suppressChange = true;
-  var saved = {}, editor = paneState.editor, session = editor.session;
-  saved.selection = session.selection.toJSON()
-  saved.atend = session.selection.getCursor().row >= session.getLength() - 1;
-  saved.folds = session.getAllFolds().map(function(fold) {
```

Comment:
No, when changing languages we shouldn't keep selections and things like that.  There should probably be an additional argument to this function to control whether it can discard selection.  Switching language would set it to "true".